tests/test_hubspot_newfw_all_fields.py

```diff
@@ -0,0 +1,112 @@
+            streams.append(stream_to_run_last)
+
+        for stream in streams:
+            # Get all records
+            if stream == 'contacts_by_company':
+                company_ids = [company['companyId'] for company in self.expected_records['companies']]
+                self.expected_records[stream] = test_client.read(stream, parent_ids=company_ids)
+            else:
+                self.expected_records[stream] = test_client.read(stream)
+
+        for stream, records in self.expected_records.items():
+            LOGGER.info("The test client found %s %s records.", len(records), stream)
+
+        super().setUp()
+        self.convert_datatype(self.expected_records)
+
+    def convert_datatype(self, expected_records):
+        # Convert the time stamp data type, Get keys with data and with no data 
+        self.keys_with_data={}
+        for stream, records in expected_records.items():
+            expected_keys = set()
+            for record in records:
+
+                expected_keys = expected_keys.union(record.keys())
+                # convert timestamps to string formatted datetime
+                timestamp_keys = {'timestamp'}
+                for key in timestamp_keys:
+                    timestamp = record.get(key)
+                    if timestamp:
+                        record[key]=self.datetime_from_timestamp(timestamp/1000, str_format=self.BASIC_DATE_FORMAT)
+
+            self.keys_with_data[stream] = expected_keys
+            self.keys_with_no_data[stream] = self.selected_fields.get(stream).difference(expected_keys)
+
+        return expected_records
+
+    def remove_bad_keys(self, stream):
+        # NB: The following woraround is for dynamic fields on the `deals` stream that we just can't track.
+        #     At the time of implementation there is no customer feedback indicating that these dynamic fields
+        #     would prove useful to an end user. The ones that we replicated with the test client are specific
+        #     to our test data. We have determined that the filtering of these fields is an expected behavior.
+        # deals workaround for 'property_hs_date_entered_<property>' fields
+        if stream == 'deals':
+            bad_key_prefixes = {'property_hs_date_entered_', 'property_hs_date_exited_', 'property_hs_time_in'}
+            bad_keys = set()
+            for key in self.expected_all_keys:
+                for bad_prefix in bad_key_prefixes:
+                   if key.startswith(bad_prefix):
+                        bad_keys.add(key)
+            for key in self.fields_replicated:
+                for bad_prefix in bad_key_prefixes:
+                   if key.startswith(bad_prefix):
+                        bad_keys.add(key)
+
+            for key in bad_keys:
+                if key in self.expected_all_keys:
+                    self.expected_all_keys.remove(key)
+                if key in self.fields_replicated:
+                    self.fields_replicated.remove(key)
```

Comment:
Two comments:

- I'm not sure why we need to build bad keys and then go through them.  It appears we could just remove them at the same time.
- If the key is in both the expected_all_keys and the fields_replicated this would pass.  So I am uncertain as to why we would need to remove this from both our expectations and our actual results.  Can we log if a key is removed and where it is removed from when we do this.  I would expect it to only be in the actual results (fields_replicated) and not necessary in the expectations. If this isn't a good assumption I would like to figure out what I'm not understanding.

Reply:
For the first comment, I felt the same and removed it directly instead of adding to bad_keys, but it complained that the set has changed while iterating.
For the second comment - let me print and see what is being removed. It could be the result of adding that extra fields in the above comment.

Reply:
Fixed the method to include only the bad keys that are not in both the lists.